src/utils/BranchlessMath.sol

```diff
@@ -40,13 +40,67 @@ library BranchlessMath {
 
     /**
      * @dev If `condition` is true returns `a`, otherwise returns `b`.
+     * see `BranchlessMath.ternary`
      */
-    function select(bool condition, address a, address b) internal pure returns (address) {
-        return address(uint160(select(condition, uint256(uint160(a)), uint256(uint160(b)))));
+    function ternary(bool condition, address a, address b) internal pure returns (address r) {
+        assembly {
+            r := xor(b, mul(xor(a, b), condition))
+        }
+    }
+
+    /**
+     * @dev If `condition` is true returns `a`, otherwise returns `b`.
+     * see `BranchlessMath.ternary`
+     */
+    function ternary(bool condition, bytes32 a, bytes32 b) internal pure returns (bytes32 r) {
+        assembly {
+            r := xor(b, mul(xor(a, b), condition))
+        }
+    }
+
+    /**
+     * @dev If `condition` is true returns `a`, otherwise returns `b`.
+     * see `BranchlessMath.ternary`
+     */
+    function ternaryU128(bool condition, uint128 a, uint128 b) internal pure returns (uint128 r) {
+        assembly {
+            r := xor(b, mul(xor(a, b), condition))
+        }
+    }
```

Comment:
Yeah.. even so solidity supports operator overloading, it doesn't work with many integer types, only between different types `bytes32`, `address`, `uint*`, etc...